File: DI_IOC_Observer/DI_IOC_Observer_JAVA/src/com/kretzer/PolygonService.java
package com.kretzer;

import com.sun.javaws.exceptions.InvalidArgumentException;

import java.util.Observable;
import java.util.Observer;

/**
 * Created by jason for DI_IOC_Observer_JAVA.
 */
public class PolygonService implements Observer {

    private String serviceName;
    private AbstractPolygon polygon;

    public PolygonService(String initServiceName) throws InvalidArgumentException{
        setServiceName(initServiceName);
    }

    public void update(Observable o, Object arg) {
        System.out.println("ServiceName: " + serviceName + " -- Perimeter = " + ((AbstractPolygon) arg).getPerimeter());
    }

    public void setServiceName(String newServiceName) throws InvalidArgumentException {
        if(newServiceName == null) {
            throw new InvalidArgumentException(new String[] {"Service name must not be null."});
        }
        serviceName = newServiceName;
    }
}

File: DI_IOC_Observer/DI_IOC_Observer_PHP/PolygonService.php
<?php
 
class PolygonService implements SplObserver
{
    private $serviceName = '';
    public function __construct($initServiceName) {
        $this->setServiceName($initServiceName);
    }
    
    public function setServiceName($newServiceName) {
        if((is_null($newServiceName)) || (!is_string($newServiceName) )) {
            throw new InvalidArgumentException('ServiceName must be a string.');
        }
        $this->serviceName = $newServiceName;
    }
    
    
    
    public function update(SplSubject $polygon) {
        $perimeter = $polygon->getPerimeter();
        //now do something with perimeter -- just send to error log
        error_log("ServiceName: ".$this->serviceName." -- Perimeter = ".$perimeter);
    }
    
    
}
?>